tests/TestCase/View/Helper/HtmlHelperTest.php

```diff
@@ -1637,6 +1640,35 @@ public function testMeta()
     }
 
     /**
+     * @return array
+     */
+    public function dataMetaLinksProvider()
+    {
+        return [
+            ['canonical', ['controller' => 'posts', 'action' => 'show'], '/posts/show'],
+            ['first', ['controller' => 'posts', 'action' => 'index'], '/posts'],
+            ['last', ['controller' => 'posts', 'action' => 'index', '?' => ['page' => 10]], '/posts?page=10'],
+            ['prev', ['controller' => 'posts', 'action' => 'index', '?' => ['page' => 4]], '/posts?page=4'],
+            ['next', ['controller' => 'posts', 'action' => 'index', '?' => ['page' => 6]], '/posts?page=6']
+        ];
+    }
+
+    /**
+     * test canonical and pagination meta links
+     *
+     * @param string $type
+     * @param array $url
+     * @param string $expected_url
+     * @dataProvider dataMetaLinksProvider
+     */
+    public function testMetaLinks($type, array $url, $expected_url)
+    {
+        $result = $this->Html->meta($type, $url);
+        $expected = ['link' => ['href' => $expected_url, 'rel' => $type]];
```

Comment:
Or you could use codesniffer locally to prevent you from going down the dark path :)